incubator/hnc/internal/reconcilers/object.go

```diff
@@ -538,12 +538,35 @@ func (r *ObjectReconciler) writeObject(ctx context.Context, log logr.Logger, ins
 	var err error = nil
 	stats.WriteObject(inst.GroupVersionKind())
 	if exist {
-		log.Info("Updating")
+		log.Info("Updating object")
 		err = r.Update(ctx, inst)
+		// RoleBindings can't have their Roles changed after they're created 
+		// (see  https://kubernetes.io/docs/reference/access-authn-authz/rbac/#clusterrolebinding-example).
+		// If an RB was quickly delete and re-created in an ancestor namespace
+		// - fast enough that by the time that HNC notices, the new RB exists; or 
+		// if there's a change to the RBs when HNC isn't running - HNC could see 
+		// it as an update (not a delete + create) and attempt to update the RBs in
+		// all descendant namespaces, and this will fail. In order to handle this 
+		// case, we try to delete and re-create the rolebinding here
```

Comment:
But this code isn't active _only_ for RBs - it works on _every_ object. You should explain why we did that.

Reply:
Is it to improve the success rate of this operation?

Reply:
Updated after our offline discussion